one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -112,8 +115,13 @@ public void onError(String commandName, Exception exception) {
         String directory = payload.get("directory").asText("");
         // setting the project directory will resolve any relative paths
         try {
+            Path directoryPath = Paths.get(directory);
             if (StringUtils.isEmpty(directory)) {
                 throw new BadRequestException("Missing required directory property");
+            } else if (!directoryPath.isAbsolute()) {
+                throw new ProjectDirectoryException("Directory supplied must be an absolute path");
```

Comment:
I think it's worth including the directory here as well - e.g. 

    "Project directory must be an absolute path: " + directory